proposals/NNNN-android-manager-apis.md

```diff
@@ -0,0 +1,689 @@
+4. Create a system that is abstracted from low level API details that will allow for more dynamic changes.
+
+### Managers
+
+##### SdlManager
+
+The SDL manager will be the main, developer-facing class that will be instantiated to handle lifecycles, submanagers, and any low level API needs. It contains a builder, an ISdl instantiation, and the proxy bridge interface. All submanagers will be created upon instantiation of this manager and their lifecycle controlled by it. 
+
+```java
+public class SdlManager implements ProxyBridge.LifecycleListener {
+
+    public static class Builder {
+        SdlManager sdlManager;
+
+        public Builder(){}
+
+        public Builder setAppId(final String appId){}
+
+        public Builder setAppName(final String appName){}
+
+        public Builder setIsMediaApp(final Boolean isMediaApp){}
+
+        public Builder setLanguage(final Language hmiLanguage){}
+
+        public Builder setLockScreenConfig (final LockScreenConfig lockScreenConfig){}
+
+        public Builder setHMITypes(final Vector<AppHMIType> hmiTypes){}
+
+        /**
+         * This Object type may change with the transport refactor
+         */
+        public Builder setTransportType(BaseTransportConfig transport){}
+
+        public Builder setContext(Context context){}
+
+        public SdlManager build() {}
+    }
+
+    // MANAGER GETTERS
+
+    public FileManager getFileManager() {}
+
+    public VideoStreamingManager getVideoStreamingManager() {}
+
+    public AudioStreamManager getAudioStreamManager() {}
+
+    public ScreenManager getScreenManager() {}
+
+    public LockscreenManager getLockscreenManager() {}
+
+    public PermissionManager getPermissionManager() {}
+
+    public SystemCapabilityManager getSystemCapabilityManager() {}
+
+    // SENDING REQUESTS
+
+    public void sendRPCRequest(RPCRequest request) throws SdlException {}
+
+    public void sendSequentialRequests(final List<? extends RPCRequest> rpcs, final OnMultipleRequestListener listener) throws SdlException {}
+
+    public void sendRequests(List<? extends RPCRequest> rpcs, final OnMultipleRequestListener listener) throws SdlException {}
+
+    // LIFECYCLE / OTHER
+
+    @Override
+    public void onProxyClosed(String info, Exception e, SdlDisconnectedReason reason){}
+
+    @Override
+    public void onServiceEnded(OnServiceEnded serviceEnded){}
+
+    @Override
+    public void onServiceNACKed(OnServiceNACKed serviceNACKed){}
+
+    @Override
+    public void onError(String info, Exception e){}
+}
+```
+
+#### BaseSubManager
+
+This is an abstract class that contains common lifecycle elements and methods between submanagers. This proposal only intends to include a small amount of code for this class, however, during implementation if logic is found to be common between different managers, it will be moved into this abstract class.
+
+```java
+public abstract class BaseSubManager {
+
+	// allows better handling of sub managers by maintaining their states
```

Comment:
I believe "sub managers" should be "submanagers" to match the other references throughout the proposal.